src/datagen/generated/mekanism/assets/mekanism/lang/en_us.json

```diff
@@ -1,4 +1,64 @@
+  "advancements.mekanism.endgame.plutonium.title":"Plutonium  power",
+  "advancements.mekanism.endgame.plutonium.description":"Not polonium",
+  "advancements.mekanism.endgame.polonium.title":"Polonium power",
+  "advancements.mekanism.endgame.polonium.description":"Not plutonium",
+  "advancements.mekanism.endgame.sps.title":"Going supercritical",
+  "advancements.mekanism.endgame.sps.description":"Ready to phase shift",
+  "advancements.mekanism.qio.advanced_qio.title":"Quantum level 2",
+  "advancements.mekanism.qio.advanced_qio.description":"Higher density",
+  "advancements.mekanism.qio.basic_qio.title":"Quantum level 1",
+  "advancements.mekanism.qio.basic_qio.description":"1/4 storage",
+  "advancements.mekanism.qio.elite_qio.title":"Quantum level 3",
+  "advancements.mekanism.qio.elite_qio.description":"In the millions!",
+  "advancements.mekanism.qio.portal_qio_dash.title":"Portable peek",
+  "advancements.mekanism.qio.portal_qio_dash.description":"Behold your storage from anywhere!",
+  "advancements.mekanism.qio.qio.title":"Quantum orchestra",
+  "advancements.mekanism.qio.qio.description":"You are the conductor!",
+  "advancements.mekanism.qio.qio_dash.title":"Quantum peek",
+  "advancements.mekanism.qio.qio_dash.description":"Behold your storage",
+  "advancements.mekanism.qio.ultimate_qio.title":"Super massive",
+  "advancements.mekanism.qio.ultimate_qio.description":"Parallel universe detected",
+  "advancements.mekanism.metallurgy.casing_steel.title":"The perfect foundation",
+  "advancements.mekanism.metallurgy.casing_steel.description":"The basis for even the most advanced machines",
+  "advancements.mekanism.metallurgy.metallurgic.title":"A metallur-what now?",
+  "advancements.mekanism.metallurgy.metallurgic.description":"Craft a Metallurgic Infuser",
+  "advancements.mekanism.metallurgy.steel.title":"Industrial revolution",
+  "advancements.mekanism.metallurgy.steel.description":"Create your first steel ingot",
+  "advancements.mekanism.robit.digi.title":"What is my purpose?",
+  "advancements.mekanism.robit.digi.description":"Goodbye friend",
+  "advancements.mekanism.robit.robit.title":"Welcome to the world",
+  "advancements.mekanism.robit.robit.description":"Hello friend",
+  "advancements.mekanism.teleports.portableteleporter.title":"Portable portals",
+  "advancements.mekanism.teleports.portableteleporter.description":"Instant travel from anywhere",
+  "advancements.mekanism.teleports.quantumentangloporter.title":"Quantum entanglement",
+  "advancements.mekanism.teleports.quantumentangloporter.description":"Instant resource transport",
+  "advancements.mekanism.teleports.teleportcore.title":"Instant teleportation",
+  "advancements.mekanism.teleports.teleportcore.description":"Numerous applications!",
+  "advancements.mekanism.teleports.teleporter.title":"Portals:online",
+  "advancements.mekanism.teleports.teleporter.description":"Instant travel",
```

Comment:
Where are these being added from? They should be somewhere in the [lang provider](https://github.com/mekanism/Mekanism/blob/1.18.x/src/datagen/main/java/mekanism/client/lang/MekanismLangProvider.java)

Reply:
I haven't added them yet, just trying to get nbt to be working.

Reply:
Ah I didn't read the sub message in your commit. And in regards to creating a stack with (if I understand properly) maxed upgrades the way I would do it is probably via a helper method like this:
```java
private ItemStack getMaxedGear(ItemRegistryObject<? extends IModuleContainerItem> item) {
    ItemStack stack = item.getItemStack();
    if (stack.getItem() instanceof IModuleContainerItem container) {
        for (ModuleData<?> module : MekanismAPI.getModuleHelper().getSupported(stack)) {
            container.addModule(stack, module);
            ModuleHelper.INSTANCE.load(stack, module).setInstalledCount(module.getMaxStackSize());
        }
    }
    return stack;
}
```

And then I think you can pass that stack as the display stack to the display method instead of just passing the item.

Reply:
This should work, thanks

Reply:
hmm, .getSupported(stack) return an empty array